test/PushNotificationTests/BaseTestSuite.cpp

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include <TestDef.h>
+#include "BaseTestSuite.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Microsoft::Windows::PushNotifications;
+
+void BaseTestSuite::ClassSetup()
+{
+    ::Test::Bootstrap::Setup();
+    bool isSelfContained{ false };
+    if (SUCCEEDED(TestData::TryGetValue(L"SelfContained", isSelfContained)))
+    {
+        if (!isSelfContained)
+        {
+            ::WindowsAppRuntime::SelfContained::TestInitialize(::Test::Bootstrap::TP::WindowsAppRuntimeFramework::c_PackageFamilyName);
+        }
+    }
+}
+
+void BaseTestSuite::ClassCleanup()
+{
+    ::Test::Bootstrap::Cleanup();
+}
+
+void BaseTestSuite::MethodCleanup()
+{
+    if (m_registered)
+    {
+        PushNotificationManager::Default().UnregisterAll();
+        m_registered = false;
+    }
+}
+
+HRESULT BaseTestSuite::ChannelRequestHelper(IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& channelOperation)
+{
+    if (channelOperation.wait_for(c_timeout) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        channelOperation.Cancel();
+        return HRESULT_FROM_WIN32(ERROR_TIMEOUT); // timed out or failed
```

Comment:
No `RETURN_WIN32_ERROR(ERROR_TIMEOUT)` ?

That error info would then show up in test logs

Reply:
Yup. We should always Log and use the macro for error codes IMHO